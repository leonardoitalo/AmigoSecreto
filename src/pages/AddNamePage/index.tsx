import Footer from 'components/Footer'
import Form from 'components/Form'
import NamesList from 'components/NamesList'
import { Card, Title } from 'styles/GlobalComponents'

const AddNamePage = () => {
  return (
    <>
      <Card>
        <Title>Vamos come√ßar!</Title>
        <Form />
        <NamesList />
        <Footer />
      </Card>
    </>
  )
}

export default AddNamePage
