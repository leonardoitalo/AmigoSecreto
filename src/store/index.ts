import { Action, configureStore, ThunkAction } from '@reduxjs/toolkit'
import formReducer from 'store/Form/slice'
import drawReducer from 'store/Draw/slice'

export const store = configureStore({
  reducer: {
    form: formReducer,
    draw: drawReducer,
  },
})

export type RootState = ReturnType<typeof store.getState>
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>
export type AppDispatch = typeof store.dispatch
