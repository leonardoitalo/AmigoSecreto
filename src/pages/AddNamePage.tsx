import Form from 'components/Form'
import Header from 'components/Header'
import { useSelector } from 'react-redux'
import { RootState } from 'store'
import {
  Button,
  Card,
  IconButton,
  NamesList,
  StartGameContainer,
  Title,
} from 'styles/GlobalComponents'

const AddNamePage = () => {
  const namesList = useSelector((state: RootState) => state.form.currentList)
  return (
    <>
      <Header />
      <Card>
        <Title>Vamos come√ßar!</Title>
        <Form />
        <NamesList>
          {namesList.map((name) => (
            <li>{name}</li>
          ))}
        </NamesList>
        <StartGameContainer>
          <IconButton hideOnMobile src="images/play-icone.png" />
          <Button>Iniciar brincadeira!</Button>
          <img src="images/sacola.png" alt="" />
        </StartGameContainer>
      </Card>
    </>
  )
}

export default AddNamePage
