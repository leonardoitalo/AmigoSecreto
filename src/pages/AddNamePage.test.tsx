import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import { store } from 'store'
import { describe, expect, test, vitest } from 'vitest'
import AddNamePage from './AddNamePage'

const mockNavigate = vitest.fn()
vitest.mock('react-router-dom', () => {
  return {
    useNavigate: () => mockNavigate,
  }
})

describe('a pagina de adicionar nome', () => {
  test('deve ser renderizada corretamente', () => {
    const { container } = render(
      <Provider store={store}>
        <AddNamePage />
      </Provider>
    )
    expect(container).toMatchSnapshot()
  })
})
