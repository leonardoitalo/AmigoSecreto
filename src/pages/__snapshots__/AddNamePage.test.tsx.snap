// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`a pagina de adicionar nome > deve ser renderizada corretamente 1`] = `
<div>
  <header
    class="sc-keTIit hbuRxh"
  >
    <div
      class="sc-dpBQxM gONwmC"
    />
    <img
      alt=""
      class="sc-jwIPbr hA-dvk"
      src="images/participante.png"
    />
  </header>
  <div
    class="sc-gtLWhw kLWIDP"
  >
    <h1
      class="sc-egkSDF ggtMEJ"
    >
      Vamos come√ßar!
    </h1>
    <form
      class="sc-khLCKb bNYEiN"
    >
      <img
        class="sc-ivxoEo hoMZFq"
        src="images/pessoa.png"
      />
      <input
        class="sc-blHHSb sc-fAUdSK eTXDzb iguyGZ"
        placeholder="Insira os nomes do participantes"
        type="text"
        value=""
      />
      <button
        class="sc-blHHSb sc-dstKZu eTXDzb hMKzeZ"
        disabled=""
      >
        Adicionar
      </button>
    </form>
    <ul
      class="sc-cHqXqK IhmNn"
    />
    <footer
      class="sc-ixGGxD cpLWuJ"
    >
      <img
        class="sc-ghWlax cdTRKE"
        src="images/play-icone.png"
      />
      <button
        class="sc-blHHSb sc-dntaoT eTXDzb hbKdni"
        disabled=""
      >
        Iniciar brincadeira!
      </button>
      <img
        alt=""
        src="images/sacola.png"
      />
    </footer>
  </div>
</div>
`;
